crates/ty_python_semantic/resources/mdtest/call/builtins.md

```diff
@@ -197,3 +197,9 @@ isinstance("", t.ClassVar)  # error: [invalid-argument-type]
 isinstance("", t.Final)  # error: [invalid-argument-type]
 isinstance("", t.Any)  # error: [invalid-argument-type]
 ```
+
+## The builtin `NotImplemented` constant is not callable
+
+```py
+NotImplemented()  # error: [call-non-callable]
+```
```

Comment:
Ideally we'd give a helpful subdiagnostic telling them that they probably meant to call `NotImplementedError()` rather than `NotImplemented` (the two are very different, but are very often confused for each other). I'll pick that up as a followup, though -- it's nontrivial to implement.

Reply:
Well, I couldn't resist doing _some_ of this now:
- https://github.com/astral-sh/ruff/pull/21475

But I opened an issue for the more fiddly part of this:

- https://github.com/astral-sh/ty/issues/1571